### PR TITLE
Display info messages on post install

### PR DESCRIPTION
**Pull request by eduardopelitti**
This PR aims to provide some basic error handling logic to the autolinking scripts by identifying potential drawbacks and managing the execution flow via try/catch statements and the use of throw under specific conditions.

**Example of a successful link:** (mostly unchanged, added info of some steps which were silently completing the tasks)

Screen Shot 2020-06-01 at 17 13 16

**Example of a partial link (either the app is already linked or part of it is):**

Screen Shot 2020-06-01 at 17 14 07

**Examples of failed links (some of the steps fail):**

Screen Shot 2020-06-01 at 18 10 25

Screen Shot 2020-06-01 at 17 36 02

Screen Shot 2020-06-01 at 17 33 42

### CODE REVIEW

File: autolink/postlink/activityLinker.js [hunk 1]
```diff
@@ -1,23 +1,52 @@
...
-    if (this.activityPath) {
-      var activityContent = fs.readFileSync(this.activityPath, "utf8");
+
+    var activityContent = fs.readFileSync(this.activityPath, "utf8");
+
+    try {
       activityContent = this._extendNavigationActivity(activityContent);
+      this.extendNavigationActivity = true;
+    } catch (e) {
+      errorn("   " + e);
```
Comment (ignacioola) [inline autolink/postlink/activityLinker.js:29]: Here we are concatenating a string with an `Error` type variable, so `e` will by converted to string by doing `e.toString()`, resulting in something like `Error: $MESSAGE'.

Maybe we should be explicit in what we want to output to `errorn`, such as :
```js
      errorn("   " + e.message);
```

File: autolink/postlink/activityLinker.js [hunk 2]
```diff
@@ -1,23 +1,52 @@
...
+    }
+
+    try {
       activityContent = this._removeGetMainComponentName(activityContent);
-      fs.writeFileSync(this.activityPath, activityContent);
+      this.removeGetMainComponentName = true;
+    } catch (e) {
+      errorn("   " + e);
```
Comment (ignacioola) [inline autolink/postlink/activityLinker.js:36]: Same here

File: autolink/postlink/activityLinker.js [hunk 3]
```diff
@@ -1,23 +1,52 @@
...
       activityContent = this._extendNavigationActivity(activityContent);
+      this.extendNavigationActivity = true;
+    } catch (e) {
+      errorn("   " + e);
+    }
+
+    try {
       activityContent = this._removeGetMainComponentName(activityContent);
-      fs.writeFileSync(this.activityPath, activityContent);
+      this.removeGetMainComponentName = true;
```
Comment (ignacioola) [inline autolink/postlink/activityLinker.js:34]: This naming looks too much as the method `_removeGetMainComponentName`, I see it prone to cause confusion. Maybe we could name it `this.isGetMainComponentNameRemoved` if that's the meaning of it?

File: autolink/postlink/activityLinker.js [hunk 4]
```diff
@@ -1,23 +1,52 @@
...
+    } catch (e) {
+      errorn("   " + e);
+    }
+
+    fs.writeFileSync(this.activityPath, activityContent);
+    if (this.extendNavigationActivity && this.removeGetMainComponentName) {
```
Comment (ignacioola) [inline autolink/postlink/activityLinker.js:40]: Same for `this.extendNavigationActivity`

File: autolink/postlink/appDelegateLinker.js [hunk 5]
```diff
@@ -1,78 +1,145 @@
...
+    const toRemove = [
+      /RCTRootView\s+\*rootView((.|\r|\s)*?)];\s+/,
+      /rootView.backgroundColor((.|\r)*)];\s+/,
+      /self.window((.|\r)*)];\s+/,
+      /UIViewController\s\*rootViewController((.|\r)*)];\s+/,
+      /rootViewController\.view\s+=\s+rootView;\s+/,
+      /self.window.rootViewController\s+=\s+rootViewController;\s+/,
+      /\[self.window\s+makeKeyAndVisible];\s+/
+    ];
+    let elementsRemoved = 0;
```
Comment (ignacioola) [inline autolink/postlink/appDelegateLinker.js:116]: With this naming I would expect this variable to be an array, I would name it `elementsRemovedCount` or something alike.

File: autolink/postlink/gradleLinker.js [hunk 6]
```diff
@@ -1,44 +1,76 @@
...
 // This should be the minSdkVersion required for RNN.
-var DEFAULT_MIN_SDK_VERSION = 19
+var DEFAULT_MIN_SDK_VERSION = 19;
 
 class GradleLinker {
   constructor() {
     this.gradlePath = path.rootGradle;
+    this.setKlotinVersionSuccess = false;
+    this.setKotlinPluginDependency = false;
+    this.setMinSdkVersion = false;
```
Comment (ignacioola) [inline autolink/postlink/gradleLinker.js:15]: To be consistent should we add the suffix `success` to the two properties above?

File: autolink/postlink/gradleLinker.js [hunk 7]
```diff
@@ -1,44 +1,76 @@
...
+    } catch (e) {
+      errorn("   " + e);
+    }
+    try {
       contents = this._setMinSdkVersion(contents);
-      fs.writeFileSync(this.gradlePath, contents);
-      infon('Root build.gradle linked successfully!\n');
+      this.setMinSdkVersion = true;
+    } catch (e) {
+      errorn("   " + e);
```
Comment (ignacioola) [inline autolink/postlink/gradleLinker.js:43]: Same here.

File: autolink/postlink/activityLinker.js (hunk 3 above)
Comment (ignacioola) [reply autolink/postlink/activityLinker.js:34]: I find the pattern `${METHOD_NAME}Succes` also valid

**Comment by eduardopelitti** [comment]
Closing in favor of https://github.com/underscopeio/react-native-navigation/pull/1

**Pull request closed without merging**